### PR TITLE
Allow deleting any kind of nodes after address mode formation

### LINKED ISSUE #77152: Assertion failed 'unused->OperIs(GT_CNS_INT)'
runtime-coreclr jitstress-random
coreclr Pri1 Runtime Tests Run windows arm64 checked

https://dev.azure.com/dnceng-public/public/_build/results?buildId=54121&view=logs&j=63f2a02a-7891-5505-e9da-45b13105e16b&t=af4e7a65-4894-5021-63d1-400b0a3e36a0
https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-main-7d574e4ebff54bd58d/JIT.Regression/1/console.f4fb506d.log?helixlogtype=result

```
C:\h\w\B99D09C2\w\ABDA09BD\e>type C:\h\w\B99D09C2\w\ABDA09BD\u\SetStressModes_jitstress_random_2.cmd 
set COMPlus_TieredCompilation=0
set COMPlus_JitStress=374
...
    JIT\Regression\JitBlue\Runtime_75312\Runtime_75312\Runtime_75312.cmd [FAIL]
      
      Assert failure(PID 1852 [0x0000073c], Thread: 2244 [0x08c4]): Assertion failed 'unused->OperIs(GT_CNS_INT)' in 'Runtime_75312:Test(int):int' during 'Lowering nodeinfo' (IL size 12; hash 0x98ca5083; MinOpts)
      
          File: D:\a\_work\1\s\src\coreclr\jit\lower.cpp Line: 5421
          Image: C:\h\w\B99D09C2\p\corerun.exe
```

cc @dotnet/jit-contrib

### PR DESCRIPTION
**Pull request by BruceForstall**
This is a general solution to #77152, allowing for deleting any set of nodes that contributed to an address mode, especially those nodes on the `op1` side of a GT_MUL with a zero `op2`.

Note that for the #77152 case, this deletes all the `op1` nodes, but leaves behind a dead temp var that was created as part of long MUL rationalization.

Fixes #77152.

### CODE REVIEW

File: src/coreclr/jit/lower.cpp [hunk 1]
```diff
@@ -5399,27 +5399,16 @@ bool Lowering::TryCreateAddrMode(GenTree* addr, bool isContainable, GenTree* par
...
+        if ((unused != base) && (unused != index))
         {
             JITDUMP("Removing unused node:\n  ");
             DISPNODE(unused);
 
             BlockRange().Remove(unused);
 
-            if (unused->OperIs(GT_ADD, GT_MUL, GT_LSH))
+            for (GenTree* operand : unused->Operands())
             {
```
Comment (AndyAyersMS) [inline src/coreclr/jit/lower.cpp:5410]: Maybe add `DEBUG_DESTROY_NODE(unused);` after the `for` loop?

**Review by AndyAyersMS — APPROVED** [review]

**Comment by BruceForstall** [comment]
@AndyAyersMS Do you think this is the "better" solution over the "targeted" (maybe: hack?) https://github.com/dotnet/runtime/pull/77871?

**Comment by AndyAyersMS** [comment]
> @AndyAyersMS Do you think this is the "better" solution over the "targeted" (maybe: hack?) #77871?

Yes.

**Comment by BruceForstall** [comment]
No diffs, about 0.01% throughput regression

**Pull request merged**